tests/random_test.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+from base_test import ArkoudaTest
+from context import arkouda as ak
+
+
+class RandomTest(ArkoudaTest):
+    def test_integers(self):
+        # verify same seed gives different but reproducible arrays
+        rng = ak.random.default_rng(18)
+        first = rng.integers(-(2**32), 2**32, 10)
+        second = rng.integers(-(2**32), 2**32, 10)
+        self.assertNotEqual(first.to_list(), second.to_list())
+
+        rng = ak.random.default_rng(18)
+        same_seed_first = rng.integers(-(2**32), 2**32, 10)
+        same_seed_second = rng.integers(-(2**32), 2**32, 10)
+        self.assertEqual(first.to_list(), same_seed_first.to_list())
+        self.assertEqual(second.to_list(), same_seed_second.to_list())
+
+        # test endpoint
+        rng = ak.random.default_rng()
+        all_zero = rng.integers(0, 1, 20)
+        self.assertTrue(all(all_zero.to_ndarray() == 0))
+
+        not_all_zero = rng.integers(0, 1, 20, endpoint=True)
+        self.assertTrue(any(not_all_zero.to_ndarray() != 0))
+
+        # verify that switching dtype and function from seed is still reproducible
+        rng = ak.random.default_rng(74)
+        uint_arr = rng.integers(0, 2**32, size=10, dtype="uint")
+        float_arr = rng.uniform(-1.0, 1.0, size=5)
+        bool_arr = rng.integers(0, 1, size=20, dtype="bool")
+        int_arr = rng.integers(-(2**32), 2**32, size=10, dtype="int")
+
+        rng = ak.random.default_rng(74)
+        same_seed_uint_arr = rng.integers(0, 2**32, size=10, dtype="uint")
+        same_seed_float_arr = rng.uniform(-1.0, 1.0, size=5)
+        same_seed_bool_arr = rng.integers(0, 1, size=20, dtype="bool")
+        same_seed_int_arr = rng.integers(-(2**32), 2**32, size=10, dtype="int")
+
+        self.assertEqual(uint_arr.to_list(), same_seed_uint_arr.to_list())
+        self.assertEqual(float_arr.to_list(), same_seed_float_arr.to_list())
+        self.assertEqual(bool_arr.to_list(), same_seed_bool_arr.to_list())
+        self.assertEqual(int_arr.to_list(), same_seed_int_arr.to_list())
+
+        # verify within bounds (lower inclusive and upper exclusive)
+        rng = ak.random.default_rng()
+        bounded_arr = rng.integers(-5, 5, 1000)
+        self.assertTrue(all(bounded_arr.to_ndarray() >= -5))
+        self.assertTrue(all(bounded_arr.to_ndarray() < 5))
+
+    def test_uniform(self):
+        # verify same seed gives different but reproducible arrays
```

Comment:
The first few examples of this test seem redundant with test_integers.  Could they be removed?

Reply:
good catch! can't believe i missed that! i'll get that fixed

Reply:
done!

Reply:
Thanks!